pkg/kube/inject/inject.go

```diff
@@ -471,14 +475,34 @@ func InjectionData(sidecarTemplate, valuesConfig, version string, typeMetadata *
 			return nil, "", merr
 		}
 	}
+
+	cluster := values.GetGlobal().GetMultiCluster().GetClusterName()
+	network := values.GetGlobal().GetNetwork()
+	// for istiod-remote we use the injection URL to get the cluster and network names
+	if c, n := splitPath(path); c != "" && n != "" {
+		cluster, network = c, n
+	}
+	// explicit label takes highest precedence
+	if n, ok := metadata.Labels[labeltemp.IstioNetwork]; ok {
+		network = n
+	}
+
+	// TODO either change templates or just get network/cluster from a map
+	var valuesMap map[string]interface{}
+	if err := yaml.Unmarshal([]byte(valuesConfig), &valuesMap); err != nil {
+		return nil, "", multierror.Prefix(err, "could not convert configuration values to map:")
+	}
+
 	data := SidecarTemplateData{
+		Cluster:        cluster,
+		Network:        network,
 		TypeMeta:       typeMetadata,
 		DeploymentMeta: deploymentMetadata,
 		ObjectMeta:     metadata,
 		Spec:           spec,
 		ProxyConfig:    meshConfig.GetDefaultConfig(),
 		MeshConfig:     meshConfig,
-		Values:         values,
+		Values:         valuesMap,
```

Comment:
I think this will be super confusing - Values is used in helm templates. Can you use a different name ?

Reply:
Sticking with just the map - the proto struct doesn't work well with the templates.